include/pybind11/pytypes.h

```diff
@@ -298,6 +298,22 @@ class handle : public detail::object_api<handle> {
 
 #ifdef PYBIND11_HANDLE_REF_DEBUG
 private:
+    void throw_gilstate_error(const std::string &function_name) const {
+        fprintf(stderr,
```

Comment:
Why are we printing out to stderr at all? Do we even have gaurantee that this will be flushed at all. Why not just prepend this text to the std::runtime_error?

Reply:
We are printing out to stderr because that's the most reliable means of reporting errors that should never break.

All these errors that we are getting are on really really weird settings where iostreams, exceptions, etc, are all broken.

We can't use std::runtime_error because that's doesn't work.

fflush(stderr); is a good idea though. I'll add that everywhere.